server/utils/validation.test.ts

```diff
@@ -312,3 +313,49 @@ describe('insertGenericError', () => {
     })
   })
 })
+
+describe('reallocateError', () => {
+  it('renames a property matching the source parameter to the destination parameter', () => {
+    const error = {
+      data: {
+        'invalid-params': [
+          { propertyName: '$.apiProperty', errorType: 'someErrorType' },
+          { propertyName: '$.unrelatedProperty', errorType: 'somOtherErrorType' },
+        ],
+      },
+    }
+
+    reallocateErrors(error as SanitisedError, 'apiProperty', 'uiProperty')
+
+    expect(error).toEqual({
+      data: {
+        'invalid-params': [
+          { propertyName: '$.uiProperty', errorType: 'someErrorType' },
+          { propertyName: '$.unrelatedProperty', errorType: 'somOtherErrorType' },
+        ],
+      },
+    })
+  })
+
+  it('does nothing if the desination parameter already exists', () => {
```

Comment:
I'm a bit confused here - why do we not want the apiProperty error to be reallocated?

Reply:
ok yeah, that makes sense 👍